packages/azure/data_stream/firewall_logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -192,7 +329,21 @@ processors:
       field:
         - azure.firewall.proto
       ignore_missing: true
-
+  - lowercase:
+      field: azure.firewall.protocol
+      ignore_failure: true
+  - set:
+      field: network.transport
+      value: "{{azure.firewall.protocol}}"
+      if: "ctx?.azure?.firewall?.protocol != null && ['udp', 'tcp', 'icmp'].contains(ctx.azure.firewall.protocol) && ['AZFWDnsQuery', 'AZFWApplicationRule', 'AZFWNetworkRule', 'AZFWNatRule'].contains(ctx.json.category)"
```

```suggestion
      if: "ctx.azure?.firewall?.protocol != null && ['udp', 'tcp', 'icmp'].contains(ctx.azure.firewall.protocol) && ctx.json?.category != null && ['AZFWDnsQuery', 'AZFWApplicationRule', 'AZFWNetworkRule', 'AZFWNatRule'].contains(ctx.json.category)"
```

1. `ctx` will never be `null`. No need to check for `null`.
2. Add `null` check on `json.category`

Same for below processors(s).